### PR TITLE
alfa-html alphabet bugfixes and tests

### PR DESCRIPTION
**Pull request by NiclasHedam**
This pull request increases the test coverage in alphabet.ts remarkably (from 49.01% to 68.61%), while also fixing a few bugs discovered on the way.

### CODE REVIEW

File: packages/alfa-html/test/alphabet.spec.ts [hunk 1]
```diff
@@ -201,6 +227,94 @@ test("Can lex a simple doctype", t => {
...
+  html(t, "<!doctype html 'foo'>", [
+    {
+      type: TokenType.Doctype,
+      name: "html",
+      publicId: null,
+      systemId: null,
+      forceQuirks: true
+    }
+  ]);
+});
```
[the remark this answers is not in the document]
Comment (NiclasHedam) [reply packages/alfa-html/test/alphabet.spec.ts]: They are not. One of them uses `"`, while the other uses `'`. These invoke two separate states.

[the remark this answers is not in the document]
Comment (kasperisager) [reply packages/alfa-html/test/alphabet.spec.ts]: The assertions are fine, but the test case is duplicated.

[the remark this answers is not in the document]
Comment (NiclasHedam) [reply packages/alfa-html/test/alphabet.spec.ts]: Right, yes, that is correct.

[the remark this answers is not in the document]
Comment (kasperisager) [reply packages/alfa-html/test/alphabet.spec.ts]: https://github.com/Siteimprove/alfa/blob/6f200403b107006d0d6754796835a51892f441d0/packages/alfa-html/test/alphabet.spec.ts#L234-L274

**Pull request merged**